### DISCUSSION TITLE
Ability to undo last stage

### DISCUSSION THREAD

**Discussion by mrcnski** [magit/magit#4706] [Support] [answered]
Hello fellow GitHub user,

I was just making some little stages when, suddenly, calamity - I made one too many! Then I sat back and bethought to myself; how would I ever undo that stage? Why, I should have to search through all the staged changes and find it, first of all. And then I would have to isolate that change, which may not even be possible if there were other changes already staged for the same line.

Any thoughts, fellow programmer?

**Comment by tarsius** [comment, marked as answer]
The "wip modes" can help with that: https://magit.vc/manual/magit/Wip-Modes.html.

**Reply by mrcnski** [reply]
Thanks, that's really interesting! Seems to work well, though I did run into a bug where the latest wip commit contained all the changes in the file, instead of just the line I committed. Staging another line corrected that other commit, though - not sure what happened there.

Still, could be worth enabling (seems like there is a slowdowm though, probably caused by a commit happening in the background for each stage).
